pallets/protos/src/tests.rs

```diff
@@ -484,3 +505,74 @@ fn internal_finalize_detach_should_works() {
 		assert!(<DetachedHashes<Test>>::contains_key(PROTO_HASH));
 	});
 }
+
+
+#[test]
+fn stake_should_work() {
```

Comment:
Giovanni (@sinkingsugar), this test case currently fails. 

This is because `stake_should_work()` doesn't work when the Amount Staked by Account `PUBLIC1` is equal to the Balance of Account `PUBLIC1`. 

It only works when it is greater (i.e if the balance / minted amount is greater than the staked amount). 

So I think there's a problem with the pallet?

Reply:
No bug, It's the `transfer`'s `keep_alive` parameter.
If you transfer ALL it deletes the account basically.
(I'm not 10000% sure about the utility of keeping the "account" to be honest)

Reply:
Could you dig it a bit please?

Reply:
updates?

Reply:
If the account is kept alive, tokens can still be sent to its address.

Polkadot gives the user the option keep their account alive when they transfer: (https://wiki.polkadot.network/docs/learn-balance-transfers#keep-alive-checks)

> By default, Polkadot-JS Apps will use transfer_keep_alive, ensuring that the account you send from cannot drop below the existential deposit of 1 DOT.
> 
> However, it may be that you do not want to keep this account alive (for example, because you are moving all of your funds to a different address). In this case, click on the "keep-alive" toggle at the bottom of the modal window. The label should switch from "Transfer with account keep-alive checks" - transfer_keep_alive will be used, to "Normal transfer without keep-alive checks" - transfer extrinsic will be used. As a common use case for using normal transfers is to entirely clear out the account, a second toggle will appear if you have the keep-alive check turned off that will send all the tokens in the account, minus a transaction fee, to the destination address.

Reply:
Nice find thanks